sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/predicates.scala

```diff
@@ -249,6 +237,41 @@ trait PredicateHelper extends Logging {
   }
 }
 
+/**
+ * Helper methods for collecting and replacing aliases.
+ */
+trait AliasHelper {
```

Comment:
How about moving this trait into a new file like `AliasHelper.scala`? It looks more general now.